### PR TITLE
Add Release notes to 0.12.0 release notes

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,23 @@
...
+  - `Session.create_async_job()` to create an `AsyncJob` instance from a query id.
+  - `AsyncJob.result()` now accepts argument `result_type` to return the results in different formats.
+  - `AsyncJob.to_df()` returns a `DataFrame` built from the result of this asynchronous job.
+  - `AsyncJob.query()` returns the SQL text of the executed query.
+- `DataFrame.agg()` and `RelationalGroupedDataFrame.agg()` now accept variable-length arguments.
+- Added parameters `luffix` and `rsuffix` to `DataFram.join()` and `DataFrame.cross_join()` to conveniently rename overlapping columns.
+- Added `Table.drop_table()` so you can drop the temp table after `DataFrame.cache_result()`. Also made class `Table` a context manager for it.
+- Added `Session.use_secondary_roles()`.
+- Added functions `first_value()` and `last_value()`. (contributed by @chasleslr)
+- Add `on` as an alias for `using_columns` and `how` as an alias for `join_type` in `DataFrame.join()`.
```
Comment (sfc-gh-sfan) [inline CHANGELOG.md:14]: nit: `Added`

Comment (sfc-gh-yixie) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,23 @@
...
+## 0.12.0 (2022-10-14)
+### New Features
+- Added new APIs for async job:
+  - `Session.create_async_job()` to create an `AsyncJob` instance from a query id.
+  - `AsyncJob.result()` now accepts argument `result_type` to return the results in different formats.
+  - `AsyncJob.to_df()` returns a `DataFrame` built from the result of this asynchronous job.
+  - `AsyncJob.query()` returns the SQL text of the executed query.
+- `DataFrame.agg()` and `RelationalGroupedDataFrame.agg()` now accept variable-length arguments.
+- Added parameters `luffix` and `rsuffix` to `DataFram.join()` and `DataFrame.cross_join()` to conveniently rename overlapping columns.
+- Added `Table.drop_table()` so you can drop the temp table after `DataFrame.cache_result()`. Also made class `Table` a context manager for it.
```
[the remark this answers is not in the document]
Comment (sfc-gh-yixie) [reply CHANGELOG.md]: "automatically" may be misleading so I changed the text to something else. Let me know if you think it's bad.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,23 @@
 # Release History
+## 0.12.0 (2022-10-14)
+### New Features
+- Added new APIs for async job:
+  - `Session.create_async_job()` to create an `AsyncJob` instance from a query id.
+  - `AsyncJob.result()` now accepts argument `result_type` to return the results in different formats.
+  - `AsyncJob.to_df()` returns a `DataFrame` built from the result of this asynchronous job.
+  - `AsyncJob.query()` returns the SQL text of the executed query.
+- `DataFrame.agg()` and `RelationalGroupedDataFrame.agg()` now accept variable-length arguments.
+- Added parameters `luffix` and `rsuffix` to `DataFram.join()` and `DataFrame.cross_join()` to conveniently rename overlapping columns.
```
Comment (sfc-gh-stan) [inline CHANGELOG.md:10]:
```suggestion
- Added parameters `lsuffix` and `rsuffix` to `DataFram.join()` and `DataFrame.cross_join()` to conveniently rename overlapping columns.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,23 @@
...
+  - `AsyncJob.query()` returns the SQL text of the executed query.
+- `DataFrame.agg()` and `RelationalGroupedDataFrame.agg()` now accept variable-length arguments.
+- Added parameters `lsuffix` and `rsuffix` to `DataFram.join()` and `DataFrame.cross_join()` to conveniently rename overlapping columns.
+- Added `Table.drop_table()` so you can drop the temp table after `DataFrame.cache_result()`. `Table` is also a context manager so you can also use the `with` statement to drop the cache temp table after use.
+- Added `Session.use_secondary_roles()`.
+- Added functions `first_value()` and `last_value()`. (contributed by @chasleslr)
+- Added `on` as an alias for `using_columns` and `how` as an alias for `join_type` in `DataFrame.join()`.
+
+### Bug Fixes
+- Fixed `Session.create_dataframe()` that raised an error when `schema` names had special characters.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:17]: Change "Fixed" to  "Fixed a bug in"

**Review by sfc-gh-mabrennan — APPROVED** [review]
Thanks

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sfc-gh-sfan) [reply CHANGELOG.md]: Looks good!

**Review by sfc-gh-sfan — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,23 @@
...
+## 0.12.0 (2022-10-14)
+### New Features
+- Added new APIs for async job:
+  - `Session.create_async_job()` to create an `AsyncJob` instance from a query id.
+  - `AsyncJob.result()` now accepts argument `result_type` to return the results in different formats.
+  - `AsyncJob.to_df()` returns a `DataFrame` built from the result of this asynchronous job.
+  - `AsyncJob.query()` returns the SQL text of the executed query.
+- `DataFrame.agg()` and `RelationalGroupedDataFrame.agg()` now accept variable-length arguments.
+- Added parameters `lsuffix` and `rsuffix` to `DataFram.join()` and `DataFrame.cross_join()` to conveniently rename overlapping columns.
+- Added `Table.drop_table()` so you can drop the temp table after `DataFrame.cache_result()`. `Table` is also a context manager so you can also use the `with` statement to drop the cache temp table after use.
```
[the remark this answers is not in the document]
Comment (sfc-gh-yixie) [reply CHANGELOG.md]: Updated your comments. Thanks.

**Review by sfc-gh-jdu — APPROVED** [review]

**Pull request merged**